### PR TITLE
Rainforest add solar time functionality

### PR DESCRIPTION
**Pull request by benowen-bom**
This PR adds the functionality to the Plugin GenerateSolarTime within `improver.generate_ancillaries.generate_derived_solar_fields`. 

This PR includes some updates to existing IMPROVER functionality:
* `solar.py`: Pulls out the evaluation of solar time in `calc_solar_hour_angle` into a separate function `calc_solar_time` .

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Review by dmentipl — COMMENTED** [review]
Thanks @benowen-bom. The changes look good. I've run the tests and all pass. I've made some very minor comments.

File: improver/generate_ancillaries/generate_derived_solar_fields.py [hunk 1]
```diff
@@ -59,6 +60,9 @@
 
 DEFAULT_TEMPORAL_SPACING_IN_MINUTES = 30
 
+SOLAR_TIME_CF_NAME = "local_solar_time"
+SOLAR_TIME_BRUCE_NAME = "solar_time"
```
[the remark this answers is not in the document]
Comment (benowen-bom) [reply improver/generate_ancillaries/generate_derived_solar_fields.py]: Yeah, I've removed these (@tjtg also mentioned these to me offline).

File: improver/utilities/solar.py [hunk 2]
```diff
@@ -128,10 +131,13 @@ def calc_solar_hour_angle(
             Day of the year 0 to 365, 0 = 1st January
         utc_hour:
             Hour of the day in UTC
+        normalise:
+            Normalise hour values to be in range 0-24
 
     Returns:
-        solar_hour_angle
-            Hour angles in degrees East-West
+        solar_time:
```
[the remark this answers is not in the document]
Comment (benowen-bom) [reply improver/utilities/solar.py:138]: Good pick-up. These were present in the function I separated into two, so I've dropped this instance as well as the one for `solar_hour_angle`.

**Review by dmentipl — APPROVED** [review]
Thanks @benowen-bom for making those changes. All looks good now.

**Review by btrotta-bom — APPROVED** [review]
The calculation makes sense. I had a minor suggestion for clarity in the tests (optional).

**Review by btrotta-bom — APPROVED** [review]

**Comment by MoseleyS** [comment]
Acceptance test data merged.

**Comment by benowen-bom** [comment]
Thanks @MoseleyS!

**Pull request merged**